packages/promises/promise.pony

```diff
@@ -159,6 +160,88 @@ actor Promise[A: Any #share]
     _attach(consume attach)
     promise
 
+  fun tag add[B: Any #share = A](p: Promise[B]): Promise[(A, B)] =>
+    """
+    Add two promises into one promise that returns the result of both when
```

Comment:
I think it makes sense to update the `add` and `join` descriptions to make that clear.